opengrok-indexer/src/main/java/org/opengrok/indexer/web/Statistics.java

```diff
@@ -201,55 +201,55 @@ public synchronized void addRequestTime(String category, long v) {
         return timingAvg;
     }
 
-    public synchronized void setRequestCategories(Map<String, Long> requestCategories) {
```

Comment:
the complaint in lgtm says:
```json
        {                                                                       
          "ruleId": "com.lgtm/java-queries:java/unsynchronized-getter",         
          "ruleIndex": 2,                                                       
          "message": {                                                          
            "text": "This get method is unsynchronized, but the corresponding [set method](1) is synchronized."
          },                                                                    
          "locations": [                                                        
            {                                                                   
              "physicalLocation": {                                             
                "artifactLocation": {                                           
                  "uri": "opengrok-indexer/src/main/java/org/opengrok/indexer/web/Statistics.java",
                  "uriBaseId": "%SRCROOT%",                                     
                  "index": 4                                                    
                },                                                              
                "region": {                                                     
                  "startLine": 151,                                             
                  "startColumn": 30,                                            
                  "endLine": 151,                                               
                  "endColumn": 50,                                              
                  "charOffset": 5186,                                           
                  "charLength": 20                                              
                }                                                               
              }                                                                 
            }                                                                   
          ],                              
```
Could this have unintended side effects ?

Reply:
I've thought about it as well. But then decided to follow the advice. The reasons to do so:
- using `synchronized` in method definition is considered bad because you are exposing the lock outside - you should always have an inner lock, e.g.:
```
private final Object o = new Object();
...
public void m() {
    synchronized (o) {
        ...
    }
}
```
- the attributes are not `volatile` which means that the value can actually be cached and not propagated properly
- it seems that these methods are used for serialization/deserialization and are called only from tests